src/base/bittorrent/torrentimpl.cpp

```diff
@@ -223,6 +223,11 @@ namespace
         outVector.resize(size, defaultValue);
         return outVector;
     }
+
+    int cleanLimitValue(const int value)
+    {
+        return ((value <= 0) || (value == std::numeric_limits<int>::max())) ? 0 : value;
```

Comment:
What's special about max int number that needs to be zeroed specifically? Is it to avoid overflow or something?

Reply:
> What's special about max int number that needs to be zeroed specifically? Is it to avoid overflow or something?

This is an imitation of normalization performed by libtorrent itself, to keep cache in line with the values used by libtorrent.